src/main/java/application/SessionController.java

```diff
@@ -84,74 +95,87 @@ public AnswerQuestionResponseBean answerQuestion(@RequestBody AnswerQuestionRequ
         session.setSequenceId(formEntrySession.getSequenceId() + 1);
         sessionRepo.save(session);
         AnswerQuestionResponseBean responseBean = mapper.readValue(resp.toString(), AnswerQuestionResponseBean.class);
+        responseBean.setSequenceId(formEntrySession.getSequenceId() + 1);
+        log.info("Answer response: " + responseBean);
         return responseBean;
-
     }
 
     @RequestMapping(value = Constants.URL_CURRENT, method = RequestMethod.GET)
     @ResponseBody
     public CurrentResponseBean getCurrent(@RequestBody CurrentRequestBean currentRequestBean) throws Exception {
-        SerializableSession serializableSession = sessionRepo.find(currentRequestBean.getSessionId());
-        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
-        return new CurrentResponseBean(formEntrySession);
+        log.info("Current request: " + currentRequestBean);
+        SerializableFormSession serializableFormSession = sessionRepo.find(currentRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableFormSession);
+        CurrentResponseBean currentResponseBean = new CurrentResponseBean(formEntrySession);
+        log.info("Current response: " + currentResponseBean);
+        return currentResponseBean;
     }
 
     @RequestMapping(value = Constants.URL_SUBMIT_FORM, method = RequestMethod.POST)
     @ResponseBody
     public SubmitResponseBean submitForm(@RequestBody SubmitRequestBean submitRequestBean) throws Exception {
-        SerializableSession serializableSession = sessionRepo.find(submitRequestBean.getSessionId());
-        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+        log.info("Submit form with bean: " + submitRequestBean);
+        SerializableFormSession serializableFormSession = sessionRepo.find(submitRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableFormSession);
         FormRecordProcessorHelper.processXML(formEntrySession.getSandbox(), formEntrySession.submitGetXml());
-        return new SubmitResponseBean(formEntrySession);
+        SubmitResponseBean submitResponseBean = new SubmitResponseBean(formEntrySession);
+        log.info("Submit response bean: " + submitResponseBean);
```

Comment:
This kind of logging can be done with a Servlet Filter:
* http://stackoverflow.com/questions/2158647/logging-response-body-html-from-httpservletresponse-using-spring-mvc-handlerin
* https://mdeinum.wordpress.com/2015/07/01/spring-framework-hidden-gems/